github/tests/Framework.py

```diff
@@ -117,7 +121,13 @@ def close(self):
         return self.__cnx.close()
 
     def __writeLine(self, line):
-        self.__file.write(line + "\n")
+        if atLeastPython3:
```

Comment:
I am not sure, is there a cleaner way to handle bytes/string problem here?

Reply:
That's very [EAFP spirit](https://docs.python.org/2/glossary.html#term-eafp), so in my mind that's clean :)
Other solution IMHO would be unnecessary complicated, involving `six` or `try`/`except` at the import level.

Or a variant:
```python
try:
    self.__file.write(line.encode('utf-8'))
except AttributeError: # If encode is not available, being it's Py3 it means it's bytes
    self.__file.write(line)
self.__file.write(b"\n")